internal/executor/executor.go

```diff
@@ -40,3 +42,57 @@ func (e *defaultExecutor) RunCommand(executable string, args []string, cwd util.
 	}
 	return stdout.Bytes(), stderrBuf.Bytes(), err
 }
+
+// createTempRScript creates a temporary file with R code and returns the file path.
+// The caller is responsible for removing the temporary file.
+func createTempRScript(content string) (string, func(), error) {
+	tempFile, err := os.CreateTemp("", "*")
+	if err != nil {
+		return "", nil, fmt.Errorf("failed to create temporary R script: %w", err)
+	}
+
+	// Write the R code to the temporary file
+	if _, err := tempFile.WriteString(content); err != nil {
+		tempFile.Close()
+		os.Remove(tempFile.Name())
+		return "", nil, fmt.Errorf("failed to write to temporary R script: %w", err)
+	}
+	if err := tempFile.Sync(); err != nil {
+		tempFile.Close()
+		os.Remove(tempFile.Name())
+		return "", nil, fmt.Errorf("failed to flush temporary R script to disk: %w", err)
+	}
+
+	// Return the file path and a cleanup function
+	cleanup := func() {
+		tempFile.Close()
+		os.Remove(tempFile.Name())
+	}
+	return tempFile.Name(), cleanup, nil
+}
```

Comment:
I was a little surprised I couldn't find a function that did this in the standard library / the filesystem packages we have. But maybe I just missed it. I'm happy to 🚮 this code if there's a better + more standard thing we can just use instead of not-invented-hereing this.